### PR TITLE
Separate `trust_remote_code` args

### PR DESCRIPTION
**Pull request by kylesayrs**
## Purpose ##
* Allow users to trust remote code of datasets
* Fix some of the finetune tests (all except `tests/llmcompressor/transformers/finetune/test_oneshot_then_finetune`)

## Changes ##
* Renamed `ModelArgs.trust_remote_code` to `ModelArgs.trust_remote_code_model` to avoid name conflict during arg parsing
* Added `DataTrainingArgs.trust_remote_code_data`
* Added `trust_remote_code_data` arg to tests

## Testing ##
* Removed dataset cache and ran `python3 -m pytest tests/llmcompressor/transformers/finetune`

### CODE REVIEW

**Comment by dsikka** [comment]
This LGTM and seems to only impact the finetune pathway. One suggestion @kylesayrs is to checkout a branch of https://github.com/neuralmagic/llm-compressor-testing/tree/main/.github/workflows and update such that instead of main, we checkout this branch. Can then kick off the weekly and nightly testing to make sure this doesn't cause any other impacts on args. Let me know if this is unclear.

**Comment by kylesayrs** [comment]
Initial tests were using a stale cache. In order to clear the HF dataset caches, you must remove *two* directories
```
rm -r ~/.cache/huggingface/datasets/*
rm -r ~/.cache/huggingface/modules/datasets_modules/datasets/
```

File: tests/llmcompressor/transformers/finetune/data/test_dataset_loading.py [hunk 1]
```diff
@@ -158,6 +158,7 @@ def test_datasets(self, dataset_key, dataset_config, split, do_concat):
             dataset=dataset_key,
             dataset_config_name=dataset_config,
             concatenate_data=do_concat,
+            trust_remote_code_data=True,
         )
         manager = TextGenerationDataset.load_from_registry(
             data_args.dataset,
```
Comment (dsikka) [inline tests/llmcompressor/transformers/finetune/data/test_dataset_loading.py:164]: Can we confirm this wont impact the finetune examples or nightly tests?

**Review by Satrat — COMMENTED** [review]
LGTM, but agreed with Dipika lets run the nightly and weekly tests on this change before merging since we're changing argument names

**Comment by kylesayrs** [comment]
https://github.com/neuralmagic/llm-compressor-testing/actions/runs/10823326340
https://github.com/neuralmagic/llm-compressor-testing/actions/runs/10823328643
https://github.com/neuralmagic/llm-compressor-testing/actions/runs/10823331168

**Pull request merged**